Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -580,6 +580,11 @@ void FrameLoader::stopLoading(UnloadEventPolicy unloadEventPolicy)
     if (RefPtr document = m_frame->document()) {
         // FIXME: Should the DatabaseManager watch for something like ActiveDOMObject::stop() rather than being special-cased here?
         DatabaseManager::singleton().stopDatabases(*document, nullptr);
+
+        if (document->settings().navigationAPIEnabled() && unloadEventPolicy != UnloadEventPolicy::UnloadAndPageHide) {
+            RefPtr window = m_frame->document()->domWindow();
+            window->protectedNavigation()->abortOngoingNavigationIfNeeded();
+        }
```

Comment:
Why did this code move here from stopAllLoaders?

Reply:
To access `unloadEventPolicy`

Reply:
Ah, I missed that.